ZK connection time out config variable ?

org.apache.kafka.connect.errors.ConnectException: Exiting WorkerSinkTask due to unrecoverable exception.
	at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:560)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:321)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:224)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:192)
	at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:177)
	at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:227)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.apache.solr.common.SolrException: java.util.concurrent.TimeoutException: Could not connect to ZooKeeper zookeeper.cert.scopussearch.net:2181/author within 15000 ms
	at org.apache.solr.common.cloud.SolrZkClient.<init>(SolrZkClient.java:201)
	at org.apache.solr.common.cloud.SolrZkClient.<init>(SolrZkClient.java:125)
	at org.apache.solr.common.cloud.SolrZkClient.<init>(SolrZkClient.java:115)
	at org.apache.solr.common.cloud.ZkStateReader.<init>(ZkStateReader.java:355)
	at org.apache.solr.client.solrj.impl.ZkClientClusterStateProvider.getZkStateReader(ZkClientClusterStateProvider.java:175)
	at org.apache.solr.client.solrj.impl.ZkClientClusterStateProvider.connect(ZkClientClusterStateProvider.java:160)
	at org.apache.solr.client.solrj.impl.BaseCloudSolrClient.connect(BaseCloudSolrClient.java:331)
	at org.apache.solr.client.solrj.impl.BaseCloudSolrClient.requestWithRetryOnStaleState(BaseCloudSolrClient.java:839)
	at org.apache.solr.client.solrj.impl.BaseCloudSolrClient.request(BaseCloudSolrClient.java:829)
	at org.apache.solr.client.solrj.SolrRequest.process(SolrRequest.java:211)
	at org.apache.solr.client.solrj.SolrRequest.process(SolrRequest.java:228)
	at com.github.jcustenborder.kafka.connect.solr.CloudSolrSinkTask.process(CloudSolrSinkTask.java:53)
	at com.github.jcustenborder.kafka.connect.solr.SolrSinkTask.put(SolrSinkTask.java:75)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:538)
	... 10 more
Caused by: java.util.concurrent.TimeoutException: Could not connect to ZooKeeper test.zookeeper.com/test within 15000 ms
	at org.apache.solr.common.cloud.ConnectionManager.waitForConnected(ConnectionManager.java:250)
	at org.apache.solr.common.cloud.SolrZkClient.<init>(SolrZkClient.java:192)

--- a/README.md
+++ b/README.md
@@ -1,6 +1,7 @@
-
 # Introduction
+[Documentation](https://jcustenborder.github.io/kafka-connect-documentation/projects/kafka-connect-solr)
 
+[Confluent Hub](https://www.confluent.io/hub/jcustenborder/kafka-connect-solr)
 
 The SOLR connector is a high speed mechanism for writing data to [Apache Solr](http://lucene.apache.org/solr/).
 

--- a/pom.xml
+++ b/pom.xml
@@ -23,7 +23,7 @@
     <parent>
         <groupId>com.github.jcustenborder.kafka.connect</groupId>
         <artifactId>kafka-connect-parent</artifactId>
-        <version>2.1.1-cp1</version>
+        <version>2.2.1-cp1</version>
     </parent>
     <artifactId>kafka-connect-solr</artifactId>
     <version>0.1-SNAPSHOT</version>

--- a/src/main/java/com/github/jcustenborder/kafka/connect/solr/CloudSolrSinkConnectorConfig.java
+++ b/src/main/java/com/github/jcustenborder/kafka/connect/solr/CloudSolrSinkConnectorConfig.java
@@ -25,17 +25,31 @@ class CloudSolrSinkConnectorConfig extends SolrSinkConnectorConfig {
 
   public static final String ZOOKEEPER_HOSTS_CONFIG = "solr.zookeeper.hosts";
   public static final String ZOOKEEPER_CHROOT_CONFIG = "solr.zookeeper.chroot";
-  public static final String COLLECTION_NAME_CONFIG = "solr.collection.name";
+  public static final String ZOOKEEPER_CONNECT_TIMEOUT_CONFIG = "solr.zookeeper.connect.timeout.ms";
+  public static final String ZOOKEEPER_CLIENT_TIMEOUT_CONFIG = "solr.zookeeper.client.timeout.ms";
+  public static final String ZOOKEEPER_RETRY_EXPIRY_TIME_CONFIG = "solr.zookeeper.retry.expiry.time.ms";
+
+
   private static final String ZOOKEEPER_HOSTS_DOC = "Zookeeper hosts that are used to store solr configuration.";
   private static final String ZOOKEEPER_CHROOT_DOC = "Chroot within solr for the zookeeper configuration.";
+  private static final String ZOOKEEPER_CONNECT_TIMEOUT_DOC = "Set the connect timeout to the zookeeper ensemble in ms.";
+  private static final String ZOOKEEPER_CLIENT_TIMEOUT_DOC = "Set the timeout to the zookeeper ensemble in ms.";
+  private static final String ZOOKEEPER_RETRY_EXPIRY_TIME_DOC = "This is the time to wait to refetch the " +
+      "state after getting the same state version from ZK in ms.";
 
   public final List<String> zookeeperHosts;
   public final String zookeeperChroot;
+  public final int zookeeperConnectTimeoutMs;
+  public final int zookeeperClientTimeoutMs;
+  public final int zookeeperRetryExpiryTimeMs;
 
   protected CloudSolrSinkConnectorConfig(Map<String, String> props) {
     super(config(), props);
     this.zookeeperHosts = this.getList(ZOOKEEPER_HOSTS_CONFIG);
     this.zookeeperChroot = this.getString(ZOOKEEPER_CHROOT_CONFIG);
+    this.zookeeperConnectTimeoutMs = getInt(ZOOKEEPER_CONNECT_TIMEOUT_CONFIG);
+    this.zookeeperClientTimeoutMs = getInt(ZOOKEEPER_CLIENT_TIMEOUT_CONFIG);
+    this.zookeeperRetryExpiryTimeMs = getInt(ZOOKEEPER_RETRY_EXPIRY_TIME_CONFIG);
   }
 
 
@@ -53,6 +67,27 @@ class CloudSolrSinkConnectorConfig extends SolrSinkConnectorConfig {
                 .documentation(ZOOKEEPER_CHROOT_DOC)
                 .group(CONNECTION_GROUP)
                 .defaultValue(null)
+                .build()
+        ).define(
+            ConfigKeyBuilder.of(ZOOKEEPER_CONNECT_TIMEOUT_CONFIG, ConfigDef.Type.INT)
+                .importance(ConfigDef.Importance.LOW)
+                .documentation(ZOOKEEPER_CONNECT_TIMEOUT_DOC)
+                .group(CONNECTION_GROUP)
+                .defaultValue(15000)
+                .build()
+        ).define(
+            ConfigKeyBuilder.of(ZOOKEEPER_CLIENT_TIMEOUT_CONFIG, ConfigDef.Type.INT)
+                .importance(ConfigDef.Importance.LOW)
+                .documentation(ZOOKEEPER_CLIENT_TIMEOUT_DOC)
+                .group(CONNECTION_GROUP)
+                .defaultValue(45000)
+                .build()
+        ).define(
+            ConfigKeyBuilder.of(ZOOKEEPER_RETRY_EXPIRY_TIME_CONFIG, ConfigDef.Type.INT)
+                .importance(ConfigDef.Importance.LOW)
+                .documentation(ZOOKEEPER_RETRY_EXPIRY_TIME_DOC)
+                .group(CONNECTION_GROUP)
+                .defaultValue(3000)
                 .build()
         );
   }

--- a/src/main/java/com/github/jcustenborder/kafka/connect/solr/CloudSolrSinkTask.java
+++ b/src/main/java/com/github/jcustenborder/kafka/connect/solr/CloudSolrSinkTask.java
@@ -16,7 +16,6 @@
 package com.github.jcustenborder.kafka.connect.solr;
 
 
-import org.apache.solr.client.solrj.SolrClient;
 import org.apache.solr.client.solrj.SolrServerException;
 import org.apache.solr.client.solrj.impl.CloudSolrClient;
 import org.apache.solr.client.solrj.request.UpdateRequest;
@@ -26,6 +25,7 @@ import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
 import java.util.Map;
+import java.util.concurrent.TimeUnit;
 
 public class CloudSolrSinkTask extends SolrSinkTask<CloudSolrSinkConnectorConfig> {
   private static final Logger log = LoggerFactory.getLogger(CloudSolrSinkTask.class);
@@ -35,7 +35,7 @@ public class CloudSolrSinkTask extends SolrSinkTask<CloudSolrSinkConnectorConfig
     return new CloudSolrSinkConnectorConfig(settings);
   }
 
-  SolrClient client;
+  CloudSolrClient client;
 
   @Override
   public void start(Map<String, String> settings) {
@@ -44,6 +44,9 @@ public class CloudSolrSinkTask extends SolrSinkTask<CloudSolrSinkConnectorConfig
     builder.withZkHost(this.config.zookeeperHosts);
     builder.withZkChroot(this.config.zookeeperChroot);
     this.client = builder.build();
+    this.client.setZkConnectTimeout(this.config.zookeeperConnectTimeoutMs);
+    this.client.setZkClientTimeout(this.config.zookeeperClientTimeoutMs);
+    this.client.setRetryExpiryTime((int) TimeUnit.SECONDS.convert(this.config.zookeeperRetryExpiryTimeMs, TimeUnit.MILLISECONDS));
   }
 
   @Override

